qnm_filter/utility.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Feb 20 17:55:14 2023
+
+@author: neil
+"""
+
+
+from joblib import Parallel, delayed
+import matplotlib.pyplot as pl
+import numpy as np
+import time
+
+def parallel_compute(function, x_arr, y_arr, kwargs = {}, show_time = True):
+    """Parallel computation of a function that takes 2 arguments 
+    
+    Arguments
+    ---------
+    function : function
+        function that takes 2 args and an arbitrary number of kwargs
+    x_arr : array-like
+        array of the values of the first arg of the function
+    y_arr : array-like
+        array of the values of the second arg of the function
+    kwargs : dict
+        dictionary of kwargs of the function
+    show_time : bool
+        whether to print time (in minutes) for the calculation
+        
+    Returns
+    ---------
+    reshaped_results : ndarray
+        2d array of the results with shape (len(x_arr), len(y_arr))
+    """
+    tic = time.time()
```

Comment:
The time measurement is typically not needed, at least at this code level, especially when we run the code on a cluster. So please remove all the relevant parts.